CONTRIBUTING.md

```diff
@@ -0,0 +1,82 @@
+# Welcome to HighTeach contributing guide <!-- omit in toc -->
+
+Thank you for investing your time in contributing to our project! :sparkles:
+
+In this guide you will get an overview of the contribution workflow from opening an issue, creating a PR, reviewing, and merging the PR.
+
+
+## New contributor guide
+
+To get an overview of the project, read the [README](README.md). Here are some resources to help you get started with open source contributions:
+
+- [Finding ways to contribute to open source on GitHub](https://docs.github.com/en/get-started/exploring-projects-on-github/finding-ways-to-contribute-to-open-source-on-github)
+- [Set up Git](https://docs.github.com/en/get-started/quickstart/set-up-git)
+- [GitHub flow](https://docs.github.com/en/get-started/quickstart/github-flow)
+- [Collaborating with pull requests](https://docs.github.com/en/github/collaborating-with-pull-requests)
+
+
+## Getting started
+
+### Issues
+
+#### Create a new issue
+
+If you spot a problem with HighTeach, [search if an issue already exists](https://github.com/redhat-beyond/HighTeach/issues). If a related issue doesn't exist, you can [open a new issue](https://github.com/redhat-beyond/HighTeach/issues/new/choose).
+
+#### Solve an issue
+
+Scan through our [existing issues](https://github.com/redhat-beyond/HighTeach/issues) to find one that interests you. As a general rule, we don’t assign issues to anyone. If you find an issue to work on, you are welcome to open a PR with a fix.
+
+### Make Changes
+
+1. Fork the repository.
+- Using GitHub Desktop:
+  - [Getting started with GitHub Desktop](https://docs.github.com/en/desktop/installing-and-configuring-github-desktop/getting-started-with-github-desktop) will guide you through setting up Desktop.
+  - Once Desktop is set up, you can use it to [fork the repo](https://docs.github.com/en/desktop/contributing-and-collaborating-using-github-desktop/cloning-and-forking-repositories-from-github-desktop)!
+
+- Using the command line:
+  - [Fork the repo](https://docs.github.com/en/github/getting-started-with-github/fork-a-repo#fork-an-example-repository) so that you can make your changes without affecting the original project until you're ready to merge them.
+
+2. Follow the instructions in the [README.md file](README.md) to set up a working environment.
+
+3. Create a working branch and start with your changes!
+
+### Commit your update
+
+Commit the changes once you are happy with them.
+The commit message should explain what changes have been made, and why. 
+In addition, they should include a link to the relevant `issue`.
+
+Please use the guide below in order to follow our commit message convention:
+* [commit convention](https://cbea.ms/git-commit/#seven-rules)
+
+### Pull Request
+
+When you're finished with the changes, create a pull request, also known as a PR.
+
+- Each Pull Request should focus on single repository principle.
```

Comment:
What does "repository principle" means?

Reply:
It means that we should apply the principle of single responsibility. If our code is doing more than one thing we should break it into different PRs.
I reworded it and squashed it to the previous commit that  contains all the minors updates of the CONTRIBUTING.md file.